tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php

```diff
@@ -197,4 +197,52 @@ public function test_single_file_plugin_without_error_for_trademarks() {
 		$this->assertSame( 0, $check_result->get_error_count() );
 		$this->assertSame( 0, $check_result->get_warning_count() );
 	}
+
+	public function test_run_with_errors_parser_warnings() {
+		$readme_check  = new Plugin_Readme_Check();
+		$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-parser-warnings/load.php' );
+		$check_result  = new Check_Result( $check_context );
+
+		$readme_check->run( $check_result );
+
+		$warnings = $check_result->get_warnings();
+
+		$this->assertNotEmpty( $warnings );
+		$this->assertArrayHasKey( 'readme.txt', $warnings );
+		$this->assertEquals( 1, $check_result->get_warning_count() );
+
+		// Check for parser warning.
+		$this->assertArrayHasKey( 0, $warnings['readme.txt'] );
+		$this->assertArrayHasKey( 0, $warnings['readme.txt'][0] );
+		$this->assertArrayHasKey( 'code', $warnings['readme.txt'][0][0][0] );
+		$this->assertEquals( 'readme_parser_warnings', $warnings['readme.txt'][0][0][0]['code'] );
+	}
+
+	public function test_filter_readme_warnings_ignored() {
+		// Define custom ignore for testing.
+		$custom_ignores = array(
+			'requires_php_header_ignored',
+		);
+
+		// Create a mock filter that will return our custom ignores.
+		$filter_name = 'wp_plugin_check_ignored_readme_warnings';
+		add_filter(
+			$filter_name,
+			static function () use ( $custom_ignores ) {
+				return $custom_ignores;
+			}
+		);
+
+		$result = apply_filters( 'wp_plugin_check_ignored_readme_warnings', array() );
+
+		$this->assertEquals( $custom_ignores, $result );
+
+		// Remove the filter to avoid interfering with other tests.
+		remove_filter(
+			$filter_name,
+			static function () use ( $custom_ignores ) {
+				return $custom_ignores;
+			}
+		);
+	}
```

Comment:
I propose to a test that remove the error for checks using filter. something like:

```
public function test_filter_wp_plugin_check_ignored_readme_warnings_will_return_no_error() {
	// Define custom ignore for testing.
	$custom_ignores = array(
		'requires_php_header_ignored',
		'contributor_ignored',
	);

	// Create a mock filter that will return our custom ignores.
	$filter_name = 'wp_plugin_check_ignored_readme_warnings';
	add_filter(
		$filter_name,
		static function () use ( $custom_ignores ) {
			return $custom_ignores;
		}
	);

	$readme_check  = new Plugin_Readme_Check();
	$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-parser-warnings/load.php' );
	$check_result  = new Check_Result( $check_context );

	$readme_check->run( $check_result );

	$errors   = $check_result->get_errors();
	$warnings = $check_result->get_warnings();

	$this->assertEmpty( $errors );
	$this->assertEmpty( $warnings );
	$this->assertSame( 0, $check_result->get_error_count() );
	$this->assertSame( 0, $check_result->get_warning_count() );

	// Remove the filter to avoid interfering with other tests.
	remove_filter(
		$filter_name,
		static function () use ( $custom_ignores ) {
			return $custom_ignores;
		}
	);
}
```

Reply:
Test added.